hackett-doc/scribblings/hackett/reference.scrbl

```diff
@@ -739,6 +936,188 @@ Adds the elements of @racket[xs] together and returns the sum. Equivalent to @ra
+
+@(hackett-examples
+ (zip {1 :: 2 :: 3 :: Nil} {18 :: 42 :: 50 :: Nil})
+ (zip {1 :: 2 :: 3 :: Nil} {18 :: 42 :: 50 :: 100 :: Nil})
+ (zip {1 :: 2 :: 3 :: 4 :: Nil} {18 :: 42 :: 50 :: Nil})
+ (zip (: Nil (t:List t:Integer)) (: Nil (t:List t:Integer))))}
+
+@defproc[(repeat [x a]) (t:List a)]{
+
+Returns an infinite list containing only @racket[x].
+
+@(hackett-examples
+ (take 5 (repeat 1)))}
+
+@defproc[(replicate [n Integer] [x a]) (t:List a)]{
+
+Returns a list of @racket[n] copies of @racket[x].
+
+@(hackett-examples
+ (replicate 3 1))}
+
+@defproc[(cycle! [xs (t:List a)]) (t:List a)]{
+
+Returns the infinite list @racket[{xs ++ xs ++ xs ++ ...}]. If @racket[xs] is infinite,
+@racket[{(cycle! xs)or == xs}]. This function is @tech[#:key "partial function"]{partial}, because it
+errors when given @racket[Nil].
+
+@(hackett-examples
+ (take 10 (cycle! {1 :: 2 :: 3 :: Nil}))
+ (eval:error (cycle! (: Nil (t:List t:Integer)))))}
+
+@defproc[(or [xs (t:List t:Bool)]) t:Bool]{
+
+Logically ors the elements of @racket[xs] together and returns the result. Equivalent to @racket[(foldr || False)].
+Because it uses a right fold, the only elements which will be evaluated are those before the first expression which
+evaluates to @racket[True]. Additionally, @racket[(or infinite-list)] can never return @racket[False], and
+@racket[(or (repeat False))] will never terminate.
+
+@(hackett-examples
+  (or {True :: False :: Nil})
+  (or {False :: True :: Nil})
+  (or {True :: (error! "never happens") :: Nil})
+  (or Nil))}
+
+@defproc[(and [xs (t:List t:Bool)]) t:Bool]{
+
+Logically ands the elements of @racket[xs] together and returns the result. Equivalent to @racket[(foldr && True)].
+Because it uses a right fold, the only elements which will be evaluated are those before the first expression which
+evaluates to @racket[False]. Additionally, @racket[(and infinite-list)] can never return @racket[True], and
+@racket[(and (repeat True))] will never terminate.
+
+@(hackett-examples
+  (and {True :: False :: Nil})
+  (and {False :: True :: Nil})
+  (and {False :: (error! "never happens") :: Nil})
+  (and Nil))}
+
+@defproc[(any? [p {a t:-> t:Bool}] [xs (t:List t:Bool)]) t:Bool]{
+
+Returns @racket[True] if @racket[xs] contains an element @racket[x] such that @racket[(p x)] is
+@racket[True]. Equivalent to @racket[(or (map p xs))].
+
+@(hackett-examples
+  (any? (<= 2) {1 :: 2 :: 3 :: Nil})
+  (any? (<= 4) {1 :: 2 :: 3 :: Nil})
+  (any? (<= 0) {1 :: (error! "never happens") :: Nil})
+  (any? (<= 0) Nil))}
+
+@defproc[(all? [p {a t:-> t:Bool}] [xs (t:List t:Bool)]) t:Bool]{
+
+Returns @racket[True] if for every element @racket[x] of @racket[xs] @racket[(p x)] is
+@racket[True]. Equivalent to @racket[(and (map p xs))].
+
+@(hackett-examples
+  (all? (<= 1) {1 :: 2 :: 3 :: Nil})
+  (all? (<= 2) {1 :: 2 :: 3 :: Nil})
+  (all? (<= 2) {1 :: (error! "never happens") :: Nil})
+  (all? (<= 1) Nil))}
+
+@defproc[(elem? [_ (t:Eq a)] [x a] [xs (t:List a)]) t:Bool]{
```

Comment:
I don’t think there’s currently a great way to typeset this sort of function, but I don’t think this is a good way. Maybe just use `defthing` here for now.